File: app/src/main/java/com/djavid/bitcoinrate/presenter/interfaces/TickerFragmentPresenter.java
package com.djavid.bitcoinrate.presenter.interfaces;

import com.djavid.bitcoinrate.core.Presenter;
import com.djavid.bitcoinrate.core.Router;
import com.djavid.bitcoinrate.model.heroku.Subscribe;
import com.djavid.bitcoinrate.model.heroku.Ticker;
import com.djavid.bitcoinrate.view.adapter.TickerItem;
import com.djavid.bitcoinrate.view.interfaces.TickerFragmentView;

import java.util.List;


public interface TickerFragmentPresenter extends Presenter<TickerFragmentView, Router, Object> {

    void addTickerFromServer(long token_id, long ticker_id);
    void getAllTickers(boolean refresh);
    void deleteTicker(long ticker_id);
    void loadTickerPrice(TickerItem tickerItem);
    void loadTickerPriceCMC(TickerItem tickerItem);
    void loadTickerPriceCryptonator(TickerItem tickerItem);
    List<Ticker> sortTickers(List<Ticker> tickers);
    List<Ticker> getTickers();
    List<Subscribe> getSubscribes();
}

File: app/src/main/java/com/djavid/bitcoinrate/presenter/interfaces/TickerFragmentPresenter.kt
package com.djavid.bitcoinrate.presenter.interfaces

import com.djavid.bitcoinrate.core.Presenter
import com.djavid.bitcoinrate.core.Router
import com.djavid.bitcoinrate.model.heroku.Subscribe
import com.djavid.bitcoinrate.model.heroku.Ticker
import com.djavid.bitcoinrate.view.adapter.TickerItem
import com.djavid.bitcoinrate.view.interfaces.TickerFragmentView


interface TickerFragmentPresenter : Presenter<TickerFragmentView, Router, Any> {
    val tickers: List<Ticker>
    val subscribes: List<Subscribe>

    fun addTickerFromServer(token_id: Long, ticker_id: Long)
    fun getAllTickers(refresh: Boolean)
    fun deleteTicker(ticker_id: Long)
    fun loadTickerPrice(tickerItem: TickerItem)
    fun loadTickerPriceCMC(tickerItem: TickerItem)
    fun loadTickerPriceCryptonator(tickerItem: TickerItem)
    fun sortTickers(tickers: List<Ticker>): List<Ticker>
}
